test/mocks/grpc/mocks.h

```diff
@@ -90,6 +90,8 @@ class MockAsyncClient : public RawAsyncClient {
                const Http::AsyncClient::StreamOptions& options));
 
   std::unique_ptr<testing::NiceMock<Grpc::MockAsyncRequest>> async_request_;
+  // Keep track of the number of requests to detect potential race condition.
+  int send_cnt_{};
```

Comment:
nit: I'd make this `sent_count_` per the style guide (https://google.github.io/styleguide/cppguide.html#General_Naming_Rules)